AppCenterAuth/AppCenterAuth/MSAuth.m

```diff
@@ -184,10 +185,18 @@ - (void)signIn {
     return;
   }
   if (self.clientApplication == nil || self.authConfig == nil) {
-    [self completeSignInWithErrorCode:MSACAuthErrorSignInBackgroundOrNotConfigured
-                           andMessage:@"signIn is called while it's not configured or not in the foreground."];
+    if (!self.downloadConfigInProgress) {
+      [self completeSignInWithErrorCode:MSACAuthErrorSignInBackgroundOrNotConfigured
+                             andMessage:@"signIn is called while it's not configured or not in the foreground."];
+    } else {
+      MSLogDebug([MSAppCenter logTag], @"Downloading configuration in process. Waiting it before sign-in.");
```

Comment:
Since there is no callback on finish, I suppose we should change message a bit: waiting means we are waiting for it to finish. But I see return further on.

Reply:
Yes, we are waiting. CompletionCallback is not deleted, it is usede after download is finished.